### PR TITLE
Skip updates on parent Devices cgroup

### PR DESCRIPTION
**Pull request by dubstack**
temporary fix for #932

I am currently working on introducing a cgroup hierarchy in the node in Kubernetes. See (kubernetes/kubernetes#27204)

Allowing or denying all devices by writing 'a' to devices.allow or devices.deny is
not possible once the device cgroups has children. As a libcontainer user I should atleast have the option of skipping updates on devices cgroup.

cc @cyphar @vishh @derekwaynecarr @mrunalp 
Signed-off-by: Buddha Prakash <buddhap@google.com>

### CODE REVIEW

File: libcontainer/cgroups/fs/devices_test.go [hunk 1]
```diff
@@ -66,7 +66,8 @@ func TestDevicesSetDeny(t *testing.T) {
 		"devices.allow": "a",
 	})
 
-	helper.CgroupData.config.Resources.AllowAllDevices = true
+	allowAllDevices := true
+	helper.CgroupData.config.Resources.AllowAllDevices = &allowAllDevices
 	helper.CgroupData.config.Resources.DeniedDevices = deniedDevices
 	devices := &DevicesGroup{}
 	if err := devices.Set(helper.CgroupPath, helper.CgroupData.config); err != nil {
```
Comment (hqhq) [inline libcontainer/cgroups/fs/devices_test.go]: Can you add a test that `AllowAllDevices` is nil, and make sure it won't change the old data?

**Comment by dubstack** [comment]
@hqhq I have added a test. PTAL. 

cc @mrunalp @derekwaynecarr @cyphar

File: libcontainer/configs/cgroup_unix.go [hunk 2]
```diff
@@ -36,7 +36,7 @@ type Cgroup struct {
 type Resources struct {
 	// If this is true allow access to any kind of device within the container.  If false, allow access only to devices explicitly listed in the allowed_devices list.
 	// Deprecated
-	AllowAllDevices bool `json:"allow_all_devices,omitempty"`
```
Comment (vishh) [inline libcontainer/configs/cgroup_unix.go]: Why did you drop the `omitempty` tag?

Comment (dubstack) [inline libcontainer/configs/cgroup_unix.go]: I thought we would want to omit the field from the object if its Nil ??

Comment (vishh) [inline libcontainer/configs/cgroup_unix.go]: To `omit` a field you need to specify `omitempty`.

Comment (dubstack) [inline libcontainer/configs/cgroup_unix.go]: Oh no. BrainFart. I will fix that in a sec.

File: libcontainer/cgroups/fs/devices_test.go (hunk 1 above)
Comment (dubstack) [inline libcontainer/cgroups/fs/devices_test.go]: @hqhq  Added test. PTAL.

**Comment by hqhq** [comment]
LGTM, thanks.

**Comment by dubstack** [comment]
@mrunalp @cyphar PTAL.

File: libcontainer/cgroups/fs/devices_test.go [hunk 3]
```diff
@@ -56,6 +56,19 @@ func TestDevicesSetAllow(t *testing.T) {
...
+	helper.CgroupData.config.Resources.AllowAllDevices = nil
+	if err := devices.Set(helper.CgroupPath, helper.CgroupData.config); err != nil {
+		t.Fatal(err)
+	}
+	value, err = getCgroupParamString(helper.CgroupPath, "devices.allow")
+	if err != nil {
+		t.Fatalf("Failed to parse devices.allow - %s", err)
+	}
+	if value != allowedList {
+		t.Fatal("Got the wrong value, set devices.allow failed.")
```
Comment (cyphar) [inline libcontainer/cgroups/fs/devices_test.go]: Please make this error message descriptive of the actual error. Something like `AllowAllDevices = nil changed devices policy`.

Comment (dubstack) [inline libcontainer/cgroups/fs/devices_test.go]: Makes sense! will make the change.

**Comment by dubstack** [comment]
@cyphar Updated the error message. PTAL.

**Comment by cyphar** [comment]
LGTM

/cc @opencontainers/runc-maintainers

**Comment by hqhq** [comment]
LGTM
